lib/components/mixins/hitbox.dart

```diff
@@ -0,0 +1,74 @@
+import '../position_component.dart';
+import '../../collision_detection.dart' as collision_detection;
+import '../../extensions/vector2.dart';
+
+mixin Hitbox on PositionComponent {
+  List<Vector2> _shape;
```

Comment:
I think this should be an Overlapable.
Then you can create a Polygon class that extends Overlapable and has a List<Vector2>
We will need to implement some more methods but that allows you to have more complex shapes (circles, unions) and allows you to check overlap between two shapes. containsPoint is useful but it would also be useful to inherit the overlaps method.
We can also rename Overlapable to Shape if that makes more sense

Reply:
Makes sense, I'll do the migration and fixes of `flame_geom` in the follow up branch that already has a lot of code so that the merge conflict isn't that massive. And then I'll point that PR towards this branch.